src/ucs/sys/sock.h

```diff
@@ -30,10 +30,23 @@ BEGIN_C_DECLS
 #define UCS_SOCKET_INET6_PORT(_addr) (((struct sockaddr_in6*)(_addr))->sin6_port)
 
 
-/* Returns an error if the default error handling should be
- * done (the error value will be returned to a caller),
- * otherwise `UCS_OK` */
-typedef ucs_status_t (*ucs_socket_io_err_cb_t)(void *arg, int io_errno);
+/**
+ * Error callback to handle errno and status of a given socket IO operation.
+ *
+ * @param [in] arg       User's argument for the error callback.
+ * @param [in] io_status Status set for a given IO operation.
+ *
+ * @return UCS_OK if error handling was done in the callback and no other
+ *         actions are required from a caller (UCS_ERR_CANCELED will be
+ *         returned as the result of the IO operation), UCS_ERR_NO_PROGRESS
```

Comment:
not clear: you mean if io_status is CANCELED than the callback should return UCS_OK?
need to phrase it to describe what is expected from the callback behavior

Reply:
I meant that if the user's callback returns UCS_OK, it means that error handling was done in the callback and no other actions should be done - IO operation has to return `UCS_ERR_CANCELED`